src/core/listener.h

```diff
@@ -206,3 +211,28 @@ QuicListenerParamGet(
     _Out_writes_bytes_opt_(*BufferLength)
         void* Buffer
     );
+
+//
+// Indicate event to Listener.
+//
+_IRQL_requires_max_(DISPATCH_LEVEL)
+QUIC_STATUS
+QuicListenerIndicateDispatchEvent(
+    _In_ QUIC_LISTENER* Listener,
+    _Inout_ QUIC_LISTENER_EVENT* Event
+    );
+
+
+_IRQL_requires_max_(DISPATCH_LEVEL)
+void
+QuicListenerHandleDosModeStateChange(
+    _In_ QUIC_LISTENER* Listener,
+    _In_ BOOLEAN DosModeStateChange
+    );
+
+
+_IRQL_requires_max_(DISPATCH_LEVEL)
+void
+QuicLibraryEvaluateSendRetryState(
```

Comment:
This shouldn't be in the listener header. Also, I think it's not needed in any header if you resolve the comments above.